lib/llm/src/kv_router/indexer.rs

```diff
@@ -123,12 +122,19 @@ pub fn compute_block_hash_for_seq(tokens: &[u32], kv_block_size: u32) -> Vec<Loc
     tokens
         .chunks_exact(kv_block_size as usize) // Split into chunks of kv_block_size elements
         .map(|chunk| {
```

Comment:
.map(|chunk| compute_block_hash(bytemuck::cast_slice::<u32, u8>(chunk)))

Reply:
But I also wonder if we need to worry about the router running on a different arch than the worker. 

If one was amd64 and the other arm64, we might need to map to LE (or BE) for consistency.

Reply:
We also have in token.rs which is using both rayon (par_chunk) and cast_slice.